src/Cli/dotnet/commands/dotnet-workload/WorkloadUtilities.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Deployment.DotNet.Releases;
+
+namespace Microsoft.DotNet.Workloads.Workload
+{
+    internal class WorkloadUtilities
+    {
+        internal static int VersionCompare(string first, string second)
```

Comment:
Is there a reason this is a separate utility instead of being the comparison operator for `ReleaseVersion` directly? It is a bit confusing because this method seems to directly (and only) work with strings that represent `ReleaseVersion`, but the method itself doesn't indicate that without reading the code.

Reply:
I was actually pretty annoyed about this.

ReleaseVersion can't handle 4-part versions like 9.0.100.1. Version can't handle prerelease bits like 9.0.100-preview.1. That means there isn't actually anything that can handle workload set versions that _might_ have 4-part versions and _might_ have prerelease bits.

The gc failure was actually directly because ReleaseVersion doesn't work.

Reply:
So, that makes it sound like this is a separate type, such as a `WorkloadVersion` since it has different constraints compared to both `ReleaseVersion` and `Version`. My general confusion was what each string parameter contains, and it sounds like it is a "new" version type, at least in the context of what version types we already have.

Reply:
In some sense, yes, and maybe it should be. I think that would be a good refactor, actually, but I'd rather have a bunch of other things in that somehow touch on it before trying to make it.